src/main/java/io/orten/nano/util/Database.java

```diff
@@ -1,64 +1,161 @@
-                    .configure("hibernate.cfg.xml") // configures settings from hibernate.cfg.xml
-                    .build();
-            MetadataSources mds = new MetadataSources(registry);
-            Metadata md = mds.buildMetadata();
-            sessionFactory = md.buildSessionFactory();
-            //  Make sure that the service registry is destroyed on shutdown by adding a shutdown hook to the runtime
-            Runtime.getRuntime().addShutdownHook(new Thread(() ->
+  
+    private static SessionFactory sessionFactory;
+    private static void init() {
+        final StandardServiceRegistry registry = new StandardServiceRegistryBuilder()
+                .configure("hibernate.cfg.xml") // configures settings from hibernate.cfg.xml
+                .build();
+        MetadataSources mds = new MetadataSources(registry);
+        Metadata md = mds.buildMetadata();
+        sessionFactory = md.buildSessionFactory();
+        //Make sure that the service registry is destroyed on shutdown by adding a shutdown hook to the runtime
+        Runtime.getRuntime().addShutdownHook(new Thread(() ->
+        {
+            StandardServiceRegistryBuilder.destroy(registry);
+        }));
+    }
+
+    /**
+     * saves an organization object to the database
+     */
+    public static boolean saveOrganization(Organization org) {
+        if (sessionFactory == null)
+            init();
+        try (Session s = sessionFactory.openSession()) {
+            s.beginTransaction();
+            String orgID=org.organizationID;
+            if (getOrganization(orgID)==null)
             {
-                StandardServiceRegistryBuilder.destroy(registry);
-            }));
+                s.save(org);
+                s.getTransaction().commit();
+                return true;
+            }else{
+                return false;
+            }
         }
+    }
 
-        /**
-         * saving an organization object to the database
-        */
-        public static void saveObject(Object o){
-            if(sessionFactory == null)
-                init();
-            try(Session s = sessionFactory.openSession()){
-                s.beginTransaction();
-                s.save(o);
-                s.getTransaction().commit();
+    /**
+     * updates an organization object already saved in saved in the database
+     */
+    public static void updateOrganization(Organization org) {
+        if (sessionFactory == null)
+            init();
+        try (Session s = sessionFactory.openSession()) {
+            s.beginTransaction();
+            s.update(org);
+        }
+    }
+
+    /**
+     * gets one organization object from the database based on its ID
+     */
+    public static Organization getOrganization(String orgID) {
+        if (sessionFactory == null)
+            init();
+        try (Session s = sessionFactory.openSession()) {
+            List<Organization> list = new ArrayList<>();
+            s.beginTransaction();
```

Comment:
Why begin a transaction if you are not going to commit it?
in this case you don't need to begin a transaction